scripts/bdm_NC_comparisson.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+#
+if [ $# -ne 4 ]; then
+    echo "Not enough arguments arguments provided, you need to provide 3 arguments:";
+    echo "All args are boolean (0 or 1)";
+    echo "";
+    echo "If 1st Argument == 1: Install requirements for bdm";
+    echo "bash bdm_NC_comparisson.sh 1 0 0 0";
+    echo "If 2nd Argument == 1: Get 10k elements";
+    echo "bash bdm_NC_comparisson.sh 0 1 0 0";
+    echo "If 3rd Argument == 1: Compute bdm";
+    echo "bash bdm_NC_comparisson.sh 0 0 1 0";
+    echo "If 4th Argument == 1: Plots bdm vs NC";
+    echo "bash bdm_NC_comparisson.sh 0 0 0 1";
+    echo "";
+    exit 1;
+fi
+#==============================================================================
+#   Variables
+#==============================================================================
+# ==============================================================================
+# Variables
+# ==============================================================================
+REQUIREMENTS=$1;
+SHUFLE=$2;
+BDM=$3;
+PLOT=$4;
+# Default Variables
+
+# listar n de 10 k : 10 / 8 / 6 estados. 5000 it
+# fazer grafico 
+# calcular bdm
+# ==============================================================================
+# Installing Requirements
+# ==============================================================================
+
+if [[ "$REQUIREMENTS" -eq "1" ]];
+    then
+
+    echo "This task in specific use the BDM Python3 API, make sure you have it available as well as pip...";
+    pip install git+https://github.com/sztal/pybdm.git
+    pip install numpy
+
+fi
+
+# ==============================================================================
+# Get 10k results from 10, 8 and 6 states and a binary alphabet
+# ==============================================================================
+
+if [[ "$SHUFLE" -eq "1" ]];
+    then
+    cd ../resultText;
+    shuf -n 10000 10st2alp-uint128.txt |  sort -k1 -n > 10k_10st_txt;
+    shuf -n 10000 8st2alp-uint128.txt | sort -k1 -n > 10k_8st_txt;
+    shuf -n 10000 resultText/6sts2alp.txt | sort -k1 -n > 10k_6st_txt;
+    cd ../scripts;
+fi
+
+# ==============================================================================
+# Compute BDM
+# ==============================================================================
+if [[ "$BDM" -eq "1" ]];
+    then
+    cd ../resultText;
+
+    function getBDM {
+        while read p; do
+            ../tm --brief --printTape -s $2 -a $3 -i 50000 -t $p > Index_tape;
+            a=`awk '{ print $1;}' Index_tape`;
+            awk '{ print $2;}' Index_tape > tape;
+            bdm=`python bdm.py tape`;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use $(...) notation instead of legacy backticked `...`.](https://app.codacy.com/app/jorgeMFS/TMCompression/pullRequest?prid=4701004)